dss/index/bundle.py

```diff
@@ -46,38 +49,59 @@ def _read_bundle_manifest(cls, replica: Replica, fqid: BundleFQID) -> dict:
         return manifest
 
     @classmethod
-    def _read_file_infos(cls, replica: Replica, fqid: BundleFQID, manifest: JSON) -> Mapping[str, JSON]:
+    def _read_file_infos(cls, replica: Replica, fqid: BundleFQID, manifest: JSON,
+                         context: LambdaContext) -> Mapping[str, JSON]:
         handle = Config.get_blobstore_handle(replica)
         bucket_name = replica.bucket
-        files_info = manifest[BundleMetadata.FILES]
-        assert isinstance(files_info, list)
+        files_info_original = manifest[BundleMetadata.FILES]
+        assert isinstance(files_info_original, list)
+        files_info = [file for file in files_info_original if file[BundleFileMetadata.INDEXED] is True]
+        files_remaining = len(files_info)
```

Comment:
Let's eliminate this by switching to a `deque` that we actually consume from. Because the deque won't have duplicate elements in it, unlike your current files_info, its length will implicitly tell you how much work is remaining.

Reply:
Will fix